src/utilities.jl

```diff
@@ -196,19 +196,19 @@ Burn vector geometries into a raster.
 The output dataset.
 """
 function unsafe_gdalrasterize(
-        dataset::Dataset,
+        dataset::T,
         options = String[];
         dest = "/vsimem/tmp"
-    )
+    ) where T<:AbstractDataset
     options = GDAL.gdalrasterizeoptionsnew(options, C_NULL)
     usage_error = Ref{Cint}()
     result = GDAL.gdalrasterize(dest, C_NULL, dataset.ptr, options, usage_error)
     GDAL.gdalrasterizeoptionsfree(options)
-    return Dataset(result)
+    return T(result)
```

Comment:
This should still be `Dataset(result)` rather than `T(result)` (see https://github.com/yeesian/ArchGDAL.jl/pull/167#issuecomment-765704084 for context).